packages/mui-base/src/TextareaAutosize/TextareaAutosize.test.tsx

```diff
@@ -28,20 +29,27 @@ describe('<TextareaAutosize />', () => {
       'ownerStatePropagation',
       'propsSpread',
       'refForwarding',
-      'rootClass',
       'slotsProp',
     ],
   }));
 
   describe('layout', () => {
     const getComputedStyleStub = {};
     function setLayout(
-      input,
-      shadow,
-      { getComputedStyle, scrollHeight, lineHeight: lineHeightArg },
+      input: HTMLTextAreaElement,
+      shadow: Element,
+      {
+        getComputedStyle,
+        scrollHeight,
+        lineHeight: lineHeightArg,
+      }: {
+        getComputedStyle: Partial<KebabKeys<CSSStyleDeclaration>>;
+        scrollHeight?: number;
+        lineHeight?: number | (() => number);
+      },
     ) {
       const lineHeight = typeof lineHeightArg === 'function' ? lineHeightArg : () => lineHeightArg;
-
+      // @ts-expect-error
       getComputedStyleStub[input] = getComputedStyle;
```

Comment:
i didn't understood why input is used as index. hence added // @ts-expect-error

Reply:
This code was previously incorrect. An HTMLElement can't be used to index an object. It will be converted to the `"[object HTMLTextAreaElement]"` string, which is pointless.
Could you please update the test to use a Map instead?

Reply:
sure.

Reply:
done